libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificateBlacklist.java

```diff
@@ -0,0 +1,225 @@
+        }
+      }
+    }
+
+    return blacklist;
+  }
+
+  boolean daySinceLastUpdate() {
+    long millisecondDiff = System.currentTimeMillis() - retrieveLastUpdateTime();
+    return millisecondDiff >= DAY_IN_MILLIS;
+  }
+
+  private long retrieveLastUpdateTime() {
+    File directory = context.getFilesDir();
+    File file = new File(directory, BLACKLIST_FILE);
+
+    long lastUpdateTime = 0;
+
+    if (file.exists()) {
+      try {
+        List<String> blacklist = obtainBlacklistContents(file);
+        lastUpdateTime = Long.valueOf(blacklist.get(BLACKLIST_HEAD));
+      } catch (IOException exception) {
+        logger.error(RETRIEVE_TIME_FAIL, exception.getMessage());
+      }
+    }
+
+    return lastUpdateTime;
+  }
+
+  void updateBlacklist() {
+    HttpUrl requestUrl = new HttpUrl.Builder().scheme(HTTPS_SCHEME)
+      .host(determineConfigEndpoint())
+      .addPathSegment("events-config")
+      .addQueryParameter(ACCESS_TOKEN_QUERY_PARAMETER, accessToken)
+      .build();
+
+    Request request = new Request.Builder()
+      .url(requestUrl)
+      .build();
+
+    OkHttpClient client = new OkHttpClient();
+    client.newCall(request).enqueue(this);
+  }
+
+  private void saveBlackList(List<String> revokedKeys) {
+    String fileContents = createListContent(revokedKeys);
+    FileOutputStream outputStream;
+
+    try {
+      outputStream = context.openFileOutput(BLACKLIST_FILE, Context.MODE_PRIVATE);
+      outputStream.write(fileContents.getBytes());
+      outputStream.close();
+    } catch (Exception exception) {
+      logger.error(SAVE_BLACKLIST_FAIL, exception.getMessage());
+    }
+  }
+
+  @Override
+  public void onFailure(Call call, IOException exception) {
+    Log.e(LOG_TAG, REQUEST_FAIL, exception);
+  }
+
+  @Override
+  public void onResponse(Call call, Response response) throws IOException {
+    List<String> revokedKeys = extractResponse(response);
+
+    saveBlackList(revokedKeys);
+  }
+
+  private String createListContent(List<String> revokedKeys) {
+    Date date = new Date();
+
+    StringBuilder content = new StringBuilder(date.getTime() + NEW_LINE);
+
+    for (String key : revokedKeys) {
+      content.append(SHA256).append(key).append(NEW_LINE);
+    }
+
+    return content.toString();
+  }
+
+  private List<String> obtainBlacklistContents(File file) throws IOException {
+    InputStream inputStream = new FileInputStream(file);
+    BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+
+    List<String> blacklist = new ArrayList<>();
+
+    try {
+      boolean done = false;
+      while (!done) {
+        String line = reader.readLine();
+        done = (line == null);
+
+        if (line != null && !line.isEmpty()) {
+          blacklist.add(line);
+        }
+      }
+    } catch (Exception exception) {
+      logger.error(READLINE_FAIL, exception.getMessage());
+    } finally {
+      reader.close();
```

Comment:
Actually, both close() methods on line 184 and 185 can potentially throw an IOException, hence they both should be enclosed into try-catch block.